server/consumer.go

```diff
@@ -692,13 +700,19 @@ func (mset *stream) addConsumerWithAssignment(config *ConsumerConfig, oname stri
 	if cName != _EMPTY_ {
 		if eo, ok := mset.consumers[cName]; ok {
 			mset.mu.Unlock()
+			if action == ActionCreate && !reflect.DeepEqual(*config, eo.config()) {
```

Comment:
Should probably `strings.ToUpper(action)` here just to be sure, otherwise the API becomes case-sensitive.

Reply:
good point.